oml/lightning/callbacks/metric.py

```diff
@@ -83,9 +124,12 @@ def calc_and_log_metrics(self, pl_module: pl.LightningModule) -> None:
         if self.log_only_main_category:
             metrics = {self.metric.overall_categories_key: metrics[self.metric.overall_categories_key]}
 
-        metrics = flatten_dict(metrics, sep="/")  # to-do: don't need
+        metrics = flatten_dict(metrics)
         pl_module.log_dict(metrics, rank_zero_only=True, add_dataloader_idx=True)
 
+        if self.save_image_logs:
+            self._log_images(pl_module=pl_module)
```

Comment:
Hm, I do not like the fact that `_log_images` requires the whole pl_module, so, it's harder to maintain proper incapsulation

What we can do is the log images in the same way as we do this for the normal metrics
now we have: 
```
self._log_images(pl_module=pl_module)
```

we can transform it to
```
plot, title = self.metric.visualise()
title = epoch_str + title  # do not take it literally, it's just a pseudo code 
self._log_images(plot, title)
```